### ISSUE TITLE
AddOriginalFile should take File and options Hash

### ISSUE THREAD

**Issue by grosscol** [samvera/hydra-works#118]
Currently the service signatue is `AddOriginalFile.call(object, path, replace=false)`
Change to `AddOriginalFile.call( object, path, options={})`

The change will more closely match the underlying call to ActiveFedora.add_file and permit caller to definemime_type, path, prefix, original_name.

**Comment by grosscol** [comment]
This issue can be more narrowly defined as the service call should accept an object which responds to read instead of a file system path.  The options hash and remainder of the function signature can be addressed after more deliberation about the use of service objects in general.

**Comment by flyingzumwalt** [comment]
Example of code that has a Rack::Upload file and could just pass that object into the services but, instead, has to break out the file's original_filename and content_type into arguments to the service.

https://github.com/projecthydra-labs/curation_concerns/blob/417bb547b6ae5fdb1f13807f6b5d442a2ccfd383/curation_concerns-models/app/actors/curation_concerns/generic_work_actor.rb#L72
```ruby
      generic_file_actor.create_content(file, file.original_filename, file.content_type)
```
which percolates down to 
https://github.com/projecthydra-labs/curation_concerns/blob/417bb547b6ae5fdb1f13807f6b5d442a2ccfd383/curation_concerns-models/app/actors/curation_concerns/generic_file_actor.rb#L55
```ruby
      Hydra::Works::UploadFileToGenericFile.call(generic_file, file.path, versioning: false, mime_type: mime_type, original_name: file_name)
```

**Comment by grosscol** [comment]
So we're going to suggest that the kinds objects passed as file to add_file will be Rack::Multipart::UploadedFile?  If the passed object does not `respond_to :mime_type` what would a default be?  I loathe to try to guess the mime_type of a chunk of bytes even if it is innocent.

My assumption was that the file passed in should minimally `respond_to :read`. What are the thoughts on checking the object for responding to mime_type and original_name, and use defaults when the methods are unavailable.

**Comment by grosscol** [comment]
Addressed by projecthydra-labs/hydra-works#154
